contracts/upgradeable_contracts/factories/HomeBridgeFactory.sol

```diff
@@ -0,0 +1,216 @@
+pragma solidity 0.4.24;
+
+import "../../IBridgeValidators.sol";
+import "../../IHomeBridge.sol";
+import "../../upgradeability/EternalStorageProxy.sol";
+import "../../upgradeability/EternalStorage.sol";
+import "../../ERC677BridgeToken.sol";
+import "../EternalOwnable.sol";
+
+contract HomeBridgeFactory is EternalStorage, EternalOwnable {
+
+  event HomeBridgeDeployed(address indexed _homeBridge, address indexed _homeValidators, address indexed _token, uint256 _blockNumber);
+
+  function getBridgeFactoryVersion() public pure returns(uint64 major, uint64 minor, uint64 patch) {
+    return (2, 2, 0);
+  }
+
+  function initialize(address _owner,
+      address _bridgeValidatorsImplementation,
+      uint256 _requiredSignatures,
+      address[] _initialValidators,
```

Comment:
On `initialize` method of `BridgeValidators`, it checks the following:
```solidity
require(_initialValidators.length >= _requiredSignatures);
```

I think it would be good to have the same check here to avoid any issues when invoking the method `deployHomeBridge`.
This check should be also added on `ForeignBridgeFactory`

Reply:
Although the `deployHomeBridge` calls the `BridgeValidators.initialize` and will fail if the wrong initial validator number and signatures requirements are set, I will add the require check to the `initialize` method and to the set `setInitialValidators` and `setRequiredSignatures` in both factories.